keylime/registrar_common.py

```diff
@@ -217,46 +277,63 @@ def do_POST(self):
             tpm_version = int(json_body['tpm_version'])
 
             # try to encrypt the AIK
-            tpm = tpm_obj.getTPM(need_hw_tpm=False,tpm_version=tpm_version)
-            (blob,key) = tpm.encryptAIK(agent_id,aik,ek,ek_tpm,aik_name)
+            tpm = tpm_obj.getTPM(need_hw_tpm=False, tpm_version=tpm_version)
+            (blob, key) = tpm.encryptAIK(agent_id, aik, ek, ek_tpm, aik_name)
             # special behavior if we've registered this uuid before
             regcount = 1
-            agent =  self.server.db.get_agent(agent_id)
+            try:
+                agent = session.query(RegistrarMain).filter_by(
+                    agent_id=agent_id).first()
+            except SQLAlchemyError as e:
+                logger.error(f'SQLAlchemy Error: {e}')
 
             if agent is not None:
 
                 # keep track of how many ek-ekcerts have registered on this uuid
-                regcount = agent['regcount']
-                if agent['ek'] != ek or agent['ekcert'] != ekcert:
-                    logger.warning('WARNING: Overwriting previous registration for this UUID with new ek-ekcert pair!')
+                regcount = agent.regcount
+                if agent.ek != ek or agent.ekcert != ekcert:
+                    logger.warning(
+                        'WARNING: Overwriting previous registration for this UUID with new ek-ekcert pair!')
                     regcount += 1
 
                 # force overwrite
                 logger.info('Overwriting previous registration for this UUID.')
-                # self.server.db.remove_agent(agent_id)
-                self.server.db.remove_agent(agent_id)
+                try:
+                    session.query(RegistrarMain).filter_by(
+                        agent_id=agent_id).delete()
+                    session.commit()
+                except SQLAlchemyError as e:
+                    logger.error(f'SQLAlchemy Error: {e}')
+
             # Add values to database
-            d={}
-            d['ek']=ek
-            d['aik']=aik
-            d['ekcert']=ekcert
-            d['virtual']=int(ekcert=='virtual')
-            d['active']=int(False)
-            d['key']=key
-            d['tpm_version']=tpm_version
```

Comment:
Are we removing the `tpm_version` dictionary entry on purpose here?

Reply:
We never retrieve this from the database, so I dropped it